### PR TITLE
tooltip support for each event

### PR DESCRIPTION
**Pull request by jpaquete**
Additional attribute in odoo xml view must be provided: tooltip="name_of_field_model"

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please change module version number and document the new feature in the README.

File: web_timeline/static/src/js/web_timeline.js [hunk 1]
```diff
@@ -98,13 +98,16 @@ odoo.define('web_timeline.TimelineView', function (require) {
...
+
             this.date_start = attrs.date_start;
             this.date_stop = attrs.date_stop;
             this.date_delay = attrs.date_delay;
             this.no_period = this.date_start == this.date_stop;
             this.zoomKey = attrs.zoomKey || '';
             this.mode = attrs.mode || attrs.default_window || 'fit';
 
+            this.tooltip = attrs.tooltip;
+
```
Comment (pedrobaeza) [inline web_timeline/static/src/js/web_timeline.js]: Put this one line above

File: web_timeline/static/src/js/web_timeline.js [hunk 2]
```diff
@@ -187,6 +190,7 @@ odoo.define('web_timeline.TimelineView', function (require) {
                    this.mode = 'fit';
                 }
             }
+
```
Comment (pedrobaeza) [inline web_timeline/static/src/js/web_timeline.js]: Don't add these extra lines (keep the diff minimal)

File: web_timeline/static/src/js/web_timeline.js [hunk 3]
```diff
@@ -248,9 +252,11 @@ odoo.define('web_timeline.TimelineView', function (require) {
                 if (eval("'" + evt[color.field] + "' " + color.opt + " '" + color.value + "'"))
                     self.color = color.color;
             });
+
             var r = {
                 'start': date_start,
                 'content': evt.__name != undefined ? evt.__name : evt.display_name,
+                'title':evt[self.tooltip],//support for tooltip
```
Comment (pedrobaeza) [inline web_timeline/static/src/js/web_timeline.js]: Add spaces for proper code style

File: web_timeline/static/src/js/web_timeline.js [hunk 4]
```diff
@@ -278,7 +284,7 @@ odoo.define('web_timeline.TimelineView', function (require) {
             }
             self.last_group_bys = n_group_bys;
             // gather the fields to get
-            var fields = _.compact(_.map(["date_start", "date_delay", "date_stop", "progress"], function (key) {
+            var fields = _.compact(_.map(["date_start", "date_delay", "date_stop", "progress", "tooltip"], function (key) {
```
Comment (pedrobaeza) [inline web_timeline/static/src/js/web_timeline.js]: AFAIK, this makes `tooltip` as a mandatory field, which is not the idea.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: web_timeline/static/src/js/web_timeline.js (hunk 4 above)
Comment (jpaquete) [reply web_timeline/static/src/js/web_timeline.js]: It does not work without that line. The attr "tooltip" is not loaded. Is there another place to to this in the code ?

Comment (pedrobaeza) [reply web_timeline/static/src/js/web_timeline.js]: You have to load the value conditionally if present

**Comment by jpaquete** [comment]
ok, review last commit. I think all is as required and also working :)

File: web_timeline/README.rst [hunk 5]
```diff
@@ -40,6 +40,8 @@ the possible attributes for the tag:
 * colors (optional): it allows to set certain specific colors if the expressed
   condition (JS syntax) is met.
```
Comment (pedrobaeza) [inline web_timeline/README.rst]: Don't introduce this new line

**Comment by pedrobaeza** [comment]
There are still comments unattended (extra lines).

**Comment by jpaquete** [comment]
check

**Comment by jpaquete** [comment]
Hi pedro please check if all ok now. Regards.

Em 26/12/2017 13:51, "João Paquete" <jpaquete@gmail.com> escreveu:

File: web_timeline/static/src/js/web_timeline.js [hunk 6]
```diff
@@ -248,9 +251,11 @@ odoo.define('web_timeline.TimelineView', function (require) {
                 if (eval("'" + evt[color.field] + "' " + color.opt + " '" + color.value + "'"))
                     self.color = color.color;
             });
+
             var r = {
                 'start': date_start,
                 'content': evt.__name != undefined ? evt.__name : evt.display_name,
+                'title': evt[self.tooltip],//support for tooltip
```
Comment (yajo) [inline web_timeline/static/src/js/web_timeline.js]: I feel the comment is a little bit useless.

**Review by yajo — APPROVED** [review]

**Review by feketemihai — APPROVED** [review]

**Comment by pedrobaeza** [comment]
My comment is still unattended: don't add extra lines, increasing the diff.

**Pull request closed without merging**